### DISCUSSION TITLE
On Fine tuning , how to mark 'unknown' word , and 'noises' and  silence ?

### DISCUSSION THREAD

**Discussion by itaipee** [openai/whisper#2121] [Q&A]
I'm working to fine-tune Whisper models to my domain , and i was wondering :   
if there unknown words in the manuals  , in my manuals there are marked by question mark    how do i mark them for the fin tuning ? 
Same question for noises ( and if there are types of noises or everything is labeled the same?

another less common is silence .  let says   the 30 seconds audio  contains 10 seconds of silence , and only then 20 seconds of speech.  
How can i mark the silence ?

**Comment by phineas-pta** [comment]
u can't

**Reply by itaipee** [reply]
so what is the best practice to deal with 'noise' , silence, and 'unknown'   tags ?     most of speech recognition data set are filled with such tags , and  usually there is no one convention for marking them.

**Reply by itaipee** [reply]
p.s   , i got an idea for reducing hallucinations that if in fine-tuning : we add few examples with silence in the beginning ,middle or end of the audio , and add distinct labels of silence accordingly. It would teach the model to avoid hallucinate and  just be ok with silence.  

although large-v2 reduced significantly the hallucinations.  large-v3 restored them :(

**Reply by dgoryeo** [reply]
@itaipee , @phineas-pta , would this approach be an option for the question: [`Whisper-AT`](https://github.com/YuanGongND/whisper-at) ?

**Reply by phineas-pta** [reply]
idk if it can help reduce hallucinations but it's worth a try

**Reply by itaipee** [reply]
interesting work is whisper-AT  ,  thank you @dgoryeo  for referring me .

my question is more simple , I want to fine-tune whisper with my data .  the manual transcribed text contains labels for noise ,and unknown words  ( tagger could not understand)     , so what are the convention for tagging such acoustic events in current Whisper ?  unknown words can be seems on different corpora  as <UNK>  ~VN  , (?),  [unknown]

**Reply by phineas-pta** [reply]
whisper doesn't tag these events nor words

**Reply by dgoryeo** [reply]
@itaipee , I second the reply from @phineas-pta.  In other words your problem description falls under the umbrella of classification, that's why Whipser-AT has extended/modified standard Whisper to produce "tagging" results.

**Comment by itaipee** [comment]
> @itaipee , @phineas-pta , would this approach be an option for the question: [`Whisper-AT`](https://github.com/YuanGongND/whisper-at) ?
